app/controllers/api_controller/providers.rb

```diff
@@ -156,5 +189,13 @@ def fetch_zone(data)
       raise BadRequestError, "Missing zone href or id" if zone_id.nil?
       resource_search(zone_id, :zone, Zone) # Only support Rbac allowed zone
     end
+
+    def role_by_configuration(configuration)
+      roles = Hash.new { |hash, key| hash[key] = key }
+      roles["bearer"] = "default"
+      endpoint = configuration["endpoint"] || configuration["authentication"] || {}
+
+      roles[endpoint["role"]] || "default"
```

Comment:
@abellotti yes, if users want to change the `default` endpoint they may omit the `role` attribute.